AppCenterDistribute/AppCenterDistribute/MSAlertController.m

```diff
@@ -168,11 +162,7 @@ + (void)makeKeyAndVisible {
     if (!windowScene) {
       windowScene = scenes.anyObject;
     }
-    
-    // FIXME: Explicit typing to call `setWindowScene:` as windowScene maybe nil and the macro won't support it.
-    SEL selector = NSSelectorFromString(@"setWindowScene:");
-    IMP impl = [window methodForSelector:selector];
-    ((void (*)(id, SEL, typeof(window))) impl)(window, selector, windowScene);
+    [MSDispatcherUtil performSelector:window withSelector:@"setWindowScene:" withObjects:@[ windowScene ]];
```

Comment:
Why are we not using `MS_DISPATCH_SELECTOR` here? Also in some weird scenario `windowScene` may be nil which would cause a crash here. I think the `FIXME` is remains unaddressed at this time.

Reply:
MS_DISPATCH_SELECTOR macro cannot accept the returned void type, so performSelector is used. We tried to make a universal macro, but it turns out to be quite large. Since it is practically not used anywhere else, we decided to simplify the implementation
Why should setWindowScene cause a crash? According to the documentation, it accepts nullable as a parameter. I checked locally and there were no errors.

Reply:
@clpolet It's what it was https://github.com/microsoft/appcenter-sdk-apple/pull/1935/commits/72d5446909be46c9088a5e1b15cbda3c3d5631e3#diff-eaebbbd7ad1e6d97b1a06ddd0cbd3677L13

Reply:
@ilyashumihin and @MatkovIvan OK yes the universal macro looked quite big 😅. I haven't tried but would it be possible to split into too macros one with return value and another one with explicit void return value. I am trying to find some coherency.

@ilyashumihin I am not worried about `setWindowScene:` itself but how we call it with possible nil parameters. If `windowScene` is nil this line crashes because it creates an array containing a nil object.

Reply:
@clpolet I made changes. Now we have 2 separate macros and the app doesn't crash when sending nil as an parameter.